internal/femto/server_test.go

```diff
@@ -150,3 +151,96 @@ func TestGetApplicationErr(t *testing.T) {
 	assert.Equal(t, http.StatusInternalServerError, w.Code)
 	assert.Contains(t, w.Body.String(), "Some application Error")
 }
+
+type TestAuthenticator struct{}
+
+func (auth TestAuthenticator) CreateToken(unit struct{}) (femto.AuthToken, error) {
+	return "token", nil
+}
+
+func (auth TestAuthenticator) RevokeToken(token femto.AuthToken) error {
+	return nil
+}
+
+func (auth TestAuthenticator) CheckToken(token femto.AuthToken) bool {
+	return token == "token"
+}
+
+func TestValidAuthentication(t *testing.T) {
```

Comment:
Worth lifting the HTTP side-effect testing scaffolding from `internal/groundstation/gpustats_test.go` to test other paths, like corrupted channels.